libazureinit/src/config.rs

```diff
@@ -696,6 +801,19 @@ mod tests {
         path = "/custom/azure-init-data-dir"
         [azure_init_log_path]
         path = "/custom/path/azure-init.log"
+        [runcmd]
+        commands = [
+            ["echo", "hello"],
+            ["apt-get", "update"]
+        ]
+        [[write_files]]
+        path = "/etc/custom-write-files"
+        owner = "test-user"
+        group = "test-user"
+        permissions = 384
```

Comment:
i assume this == 0o600, but also wonder if we should restrict it to one format to ensure correctness.

Reply:
because I would probably state permissions = 600 and be wrong.